core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/commit/CommitInfoSearchRequest.java

```diff
@@ -88,11 +100,100 @@ protected CommitInfos createEmptyResult(int limit) {
 		return new CommitInfos(limit, 0);
 	}
 	
-	private void addBranchClause(final ExpressionBuilder builder) {
+	private void addSecurityFilter(final ExpressionBuilder builder, RepositoryContext context) {
+		
+		final User user = context.service(User.class);
+		if (user.isAdministrator() || user.hasPermission(Permission.requireAll(Permission.OPERATION_BROWSE, Permission.ALL))) {
+			if (containsKey(OptionKey.BRANCH)) {
+				addBranchClause(builder, context, null, null, true);		
+			}
+			return;
+		}
+		
+		final List<Permission> readPermissions = user.getPermissions().stream()
+				.filter(p -> Permission.ALL.equals(p.getOperation()) || Permission.OPERATION_BROWSE.equals(p.getOperation()))
+				.collect(Collectors.toList());
+		
+		final Set<String> exactResourceIds = readPermissions.stream()
+				.flatMap(p -> p.getResources().stream())
+				.filter(resource -> !resource.endsWith("*"))
```

Comment:
It would be nice to introduce a helper static method to detect wildcard resources, like `Permission.isWildcardResource(String)`.